### ISSUE TITLE
HeaderMatchesChecker with regex and \r\n line endings

### ISSUE THREAD

**Issue by MichaelJAllen** [scalastyle/scalastyle#236]
Hi!

I've noticed an interesting problem when using `HeaderMatchesChecker` with regular expressions turned on and _Windows_ (`\r\n`) source file line endings. (Just to be clear, this problem does NOT occur if I disable regular expressions for this checker.)

When I'm working on _Windows_, I have _git_ check out my sources using _Window_'s native line endings (i.e. _carriage return_ + _linefeed_, or `\r\n`). My source files, as well as my `scalastyle-config.xml` config file all have these line endings.

Say my source files have the following header:
```
//======================
// Copyright © 2017, Me.
//======================
```
My `HeaderMatchesChecker` config looks like this:
 ```
<check level="warning" class="org.scalastyle.file.HeaderMatchesChecker" enabled="true">
  <parameters>
    <parameter name="regex">true</parameter>
    <parameter name="header"><![CDATA[//======================
// Copyright © \d{4}, Me\.
//======================]]></parameter>
  </parameters>
</check>
```
Unfortunately, this fails for every source file.

However, if I change the line endings of my source files to _Unix_ style (just a _linefeed_, or `\n`), then the check passes every source file!

As a workaround, I can keep my sources with _Windows_ line endings, but I need to change the `HeaderMatchesChecker` config to:
 ```
<check level="warning" class="org.scalastyle.file.HeaderMatchesChecker" enabled="true">
  <parameters>
    <parameter name="regex">true</parameter>
    <parameter name="header"><![CDATA[//======================\r?
// Copyright © \d{4}, Me\.\r?
//======================]]></parameter>
  </parameters>
</check>
```
That is, I have added an optional carriage return character at the end of the each line. The checks then work when I check the style on both _Linux_ and _Windows_.

I guess the root cause of the problem is that, when reading sources, _Scalastyle_ doesn't recognize _Windows_ line endings, and preserves any _carriage return_ character at the end of each line (instead of treating it as part of the line termination sequence and stripping it).

Interestingly, if I remove regular expressions, and do a literal check, then the sources pass muster regardless of their line endings:
```
<check level="warning" class="org.scalastyle.file.HeaderMatchesChecker" enabled="true">
  <parameters>
    <parameter name="regex">false</parameter>
    <parameter name="header"><![CDATA[//======================
// Copyright © 2017, Me.
//======================]]></parameter>
  </parameters>
</check>
```
(Unfortunately, this latter option isn't viable for me as I need to match other header content using regular expressions.)

**Comment by matthewfarwell** [comment]
This needs a bit more digging into. I don't have a windows machine handy.

**Comment by MichaelJAllen** [comment]
Hi Matthew! Thanks for the follow-up.

I think you should be OK without _Windows_.  Clearly, it will depend on your editor, but most will allow the line endings of a source file to be changed from `\n` to `\r\n` to verify this behavior - the platform just influences the default line endings employed.

For example, in _IntelliJ_, there's an indicator in the bottom right-hand corner of the window (which should read **LF** with a pair of up & down triangles). If you click on that indicator, you have the opportunity to change to a **CRLF** line ending.

Doing that should allow you to reproduce and debug the problem.

**Comment by matthewfarwell** [comment]
Fixed by #237
